### PR TITLE
Add Hide Tab Index and Hide Tab Close Button

### PR DESCRIPTION
**Pull request by pinpins**
This adds functionality to Hide Tab Index and to hide Tab Close(x) Button regardless of template used.

### CODE REVIEW

File: terminus-terminal/src/components/appearanceSettingsTab.component.pug [hunk 1]
```diff
@@ -109,6 +109,24 @@ h3.mb-3 Appearance
         (ngModelChange)='config.save()',
     )
 
+.form-line
+    .header
+        .title Disable Tab Index
+
+    toggle(
+        [(ngModel)]='config.store.terminal.disableTabIndex',
+        (ngModelChange)='config.save(); config.requestRestart()',
```
Comment (Eugeny) [inline terminus-terminal/src/components/appearanceSettingsTab.component.pug:118]: Restart shouldn't be required if you bind directly to the settings in `TabHeaderComponent`

File: terminus-core/src/components/tabHeader.component.pug [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .progressbar([style.width]='progress + "%"', *ngIf='progress != null')
-.index(
+.index(*ngIf='tab.showIndex==true',
     #handle,
     [style.background-color]='tab.color',
 ) {{index + 1}}
 .name([title]='tab.customTitle || tab.title') {{tab.customTitle || tab.title}}
-button((click)='app.closeTab(tab, true)') &times;
+button(*ngIf='tab.showCloseButton==true',(click)='app.closeTab(tab, true)') &times;
```
Comment (Eugeny) [inline terminus-core/src/components/tabHeader.component.pug:7]: You can make `config` `public` and use `!config.store.terminal.disableCloseButton` here directly

File: terminus-terminal/src/components/appearanceSettingsTab.component.pug [hunk 3]
```diff
@@ -109,6 +109,24 @@ h3.mb-3 Appearance
         (ngModelChange)='config.save()',
     )
 
+.form-line
+    .header
+        .title Disable Tab Index
```
Comment (Eugeny) [inline terminus-terminal/src/components/appearanceSettingsTab.component.pug:114]: Please use `Sentence case`

**Review by Eugeny — CHANGES_REQUESTED** [review]
Thank you for your contribution! Take a look at the linter output and I've also added a few comments here and there

**Comment by pinpins** [comment]
Thanks for review and comments, still learning your implementation for the first time. If all ok with latest commit, please cut a release, would love to use those new features.

**Comment by Eugeny** [comment]
@allcontributors please add @pinpins for code

**Pull request merged**